Mathlib/Algebra/Order/Quantale.lean

```diff
@@ -176,4 +176,26 @@ theorem rightMulResiduation_le_iff_mul_le : x ≤ y ⇨ᵣ z ↔ y * x ≤ z whe
       iSup_le_iff, implies_true]
   mpr h1 := le_sSup h1
 
+section Zero
+
+variable {α : Type*} [Semigroup α] [CompleteLattice α] [IsQuantale α]
+variable {x : α}
+
+@[to_additive (attr := simp)]
+theorem bot_mul : ⊥ * x = ⊥ := by
+  rw [← sSup_empty, sSup_mul_distrib]
+  simp only [Set.mem_empty_iff_false, not_false_eq_true, iSup_neg, iSup_bot, sSup_empty]
+
+@[to_additive (attr := simp)]
+theorem mul_bot : x * ⊥ = ⊥ := by
+  rw [← sSup_empty, mul_sSup_distrib]
+  simp only [Set.mem_empty_iff_false, not_false_eq_true, iSup_neg, iSup_bot, sSup_empty]
+
+instance : MulZeroClass α where
+  zero := ⊥
```

Comment:
This instance is dangerous since it creates data (`Zero`) out of a Prop (`IsQuantale`). You will need to assume that `⊥ = 0` (@vihdzp, did you have a mixin for that?) instead of making it defeq

Reply:
I'll remove it for the time being.

After all, there is no to_additive counterpart - and it makes sense to me to only have the theorems that work for both Mul and Add variants. I think of them as bot_mul and mul_bot anyway.